### PR TITLE
Test setup with database

### PR DESCRIPTION
**Pull request by JakobMiksch**
- uses a docker-compose file to create  a environment for the tests
- adds a basic test for PostGIS store and layer

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -171,12 +171,24 @@ describe('Datastore', () => {
     createdWorkSpace = await grc.workspaces.create(workSpace);
   });
 
-  // TODO: test PostGIS store
-  // TODO: test image mosaic
-  // TODO: test WMTS-Stores
+  it('can create a PostGIS store', async () => {
+    // TODO: read params from env (?)
+    const dataStore = 'jm-postgis';
```
Comment (JakobMiksch) [inline test/test.js:176]:
```suggestion
    const dataStore = 'my-postgis-datastore';
```

File: test/test.js [hunk 2]
```diff
@@ -171,12 +171,24 @@ describe('Datastore', () => {
     createdWorkSpace = await grc.workspaces.create(workSpace);
   });
 
-  // TODO: test PostGIS store
-  // TODO: test image mosaic
-  // TODO: test WMTS-Stores
+  it('can create a PostGIS store', async () => {
+    // TODO: read params from env (?)
```
Comment (JakobMiksch) [inline test/test.js:175]:
```suggestion
```

File: test/test.js [hunk 3]
```diff
@@ -298,6 +310,33 @@ describe('Layer', () => {
     );
   });
 
+  it('can publish a FeatureType from PostGIS', async () => {
+    // TODO: use env vars
```
Comment (JakobMiksch) [inline test/test.js:314]:
```suggestion
```

File: test/test.js [hunk 4]
```diff
@@ -298,6 +309,32 @@ describe('Layer', () => {
     );
   });
 
+  it('can publish a FeatureType from PostGIS', async () => {
```
Comment (chrismayer) [inline test/test.js:312]: This contains no test with e.g. expect. So no real test or is it hoping for no exception? Maybe adding some basic expectations would be a good thing

File: test/test.js [hunk 5]
```diff
@@ -171,12 +171,23 @@ describe('Datastore', () => {
...
+    const exposePk = true;
+
+    await grc.datastores.createPostgisStore(
+      workSpace, nameSpaceUri, dataStore, pgHost, pgPort, pgUser, pgPassword, pgSchema, pgDb, exposePk
+    );
+  });
 
   it('can create a coverage store', async () => {
-    const geotiff = 'test/sample_data/world.geotiff'
+    const geotiff = 'test/sample_data/world.tif'
```
Comment (chrismayer) [inline test/test.js:190]: Has this been renamed?

File: test/test.js [hunk 6]
```diff
@@ -384,7 +421,7 @@ describe('Layer', () => {
   })
 
   it('can create Coverage layer', async () => {
-    const geotiff = 'test/sample_data/world.geotiff'
+    const geotiff = 'test/sample_data/world.tif'
```
Comment (chrismayer) [inline test/test.js:424]: See above?

**Review by chrismayer — COMMENTED** [review]
Nice improvement! Please have a look at my comment regarding the test expectation before merging.

File: test/test.js (hunk 4 above)
Comment (JakobMiksch) [reply test/test.js:312]: fixed with e5f562c

File: test/test.js (hunk 6 above)
Comment (JakobMiksch) [reply test/test.js:424]: see above

**Review by JakobMiksch — COMMENTED** [review]
Thanks @chrismayer for the review. I addressed your remarks

File: README.md [hunk 7]
```diff
@@ -110,9 +110,7 @@ A request either succeeds or throws the custom `GeoServerResponseError`. It has
 First start a GeoServer, e.g. by using this Docker container:
```
Comment (JakobMiksch) [inline README.md:110]:
```suggestion
First start a test setup using this Docker compose file:
```

**Pull request merged**